bemani/backend/iidx/pendual.py

```diff
@@ -601,749 +682,658 @@ def handle_IIDX22music_request(self, request: Node) -> Optional[Node]:
-            elif timeshift_override == 2:
-                common_timeshift_phase.set_attribute('phase', '1')
-            elif timeshift_override == 3:
-                common_timeshift_phase.set_attribute('phase', '2')
-
-            root.add_child(Node.void('eaappli_expert'))
-
-            cc_collabo_event = Node.void('cc_collabo_event')
-            root.add_child(cc_collabo_event)
-            if ccxbm_on:
-                cc_collabo_event.set_attribute('phase', "1")
-                root.add_child(Node.void('cc_collabo_license'))
+            if cltype == self.GAME_CLTYPE_SINGLE:
+                index = self.DAN_RANKING_SINGLE
             else:
-                cc_collabo_event.set_attribute('phase', "0")
-
-            tohoisfumo = Node.void('toho_remix')
-            root.add_child(tohoisfumo)
-            tohoisfumo.set_attribute('phase', "2")
-
-            cm = Node.void("cm")
-            root.add_child(cm)
-            cm.set_attribute('compo', 'cm_event3')
-            cm.set_attribute('folder', 'cm_event3')
-            cm.set_attribute('id', "0")
-
-            # Ranking details in title screen
-            rankingcharts = []
-            for (mid, _plays) in self.data.local.music.get_hit_chart(self.game, self.music_version, 20):
-                rankingcharts.append(mid)
-            root.add_child(Node.u16_array('monthly_mranking', rankingcharts))
-            root.add_child(Node.u16_array('total_mranking', rankingcharts))
-            # need to add in the top 20 rankings for month and total network
-            # TODO: Make these separate in the future
+                index = self.DAN_RANKING_DOUBLE
 
-            return root
+            self.update_rank(
+                userid,
+                index,
+                rank,
+                percent,
+                cleared,
+                stages_cleared,
+            )
+
+        # Figure out number of players that played this ranking
+        all_achievements = self.data.local.user.get_all_achievements(self.game, self.version, achievementid=rank, achievementtype=index)
+        root = Node.void('IIDX22grade')
+        root.set_attribute('pnum', str(len(all_achievements)))
+        return root
 
-        if method == 'delete':
-            return Node.void('IIDX22pc')
+    def handle_IIDX22pc_common_request(self, request: Node) -> Optional[Node]:
+        root = Node.void('IIDX22pc')
+        root.set_attribute('expire', '600')
+
+        ir = Node.void('ir')
+        root.add_child(ir)
+        ir.set_attribute('beat', '2')
+
+        newsong_another = Node.void('newsong_another')
+        root.add_child(newsong_another)
+        newsong_another.set_attribute('open', '1')
+
+        limit = Node.void('limit')
+        root.add_child(limit)
+        limit.set_attribute('phase', '21')
+
+        # See if we configured event overrides
+        game_config = self.get_game_config()
+        if self.machine_joined_arcade():
```

Comment:
Why order change? Wouldn't get_game_config() run a DB operation that could be useless?

Reply:
This was due to the original code being bugged. `game_config` is referred to later and might not be defined. I suppose a better way of doing it would be more like
```py
        # See if we configured event overrides
        if self.machine_joined_arcade():
            game_config = self.get_game_config()
            timeshift_override = game_config.get_int('cycle_config')
            ccxbm_on = game_config.get_bool('ccxbm_enable')
            omni_events = game_config.get_bool('omnimix_events_enabled')
            event_phase = game_config.get_int('event_phase')
        else:
            # If we aren't in an arcade, we turn off events
            timeshift_override = 0
            ccxbm_on = False
            omni_events = False

        if self.omnimix and (not omni_events):
            phase = 0
        elif self.omnimix and omni_events or not self.omnimix:
            phase = event_phase
        elif not self.omnimix:
            phase = event_phase
```

Reply:
I agree that the proposed code looks better.